draft-ietf-ccwg-rfc5033bis.md

```diff
@@ -333,6 +345,24 @@ The classic congestion control algorithm {{!RFC5681}} and the widely deployed
 Cubic algorithm {{?RFC9438}} do not address it, but a new
 congestion control algorithm has the opportunity to improve the state of the art.
 
+### Protection Against High Packet Loss
+
+A congestion control algorithm should reduce its sending
+rate if experiencing high packet loss. 
```

Comment:
I would suggest expanding this, since reducing the sending rate in response to packet loss is not sufficient to protect against high packet loss.

AFAICT there are at least two parts to avoiding high packet loss:

(1) don't send unreasonably fast, based on previously available signals (e.g., don't be like Reno/CUBIC and cause 20% losses if the fair-share cwnd is 2 packets; don't slow-start with an exponential factor that causes 100x growth of the sending rate per round trip; and don't use an approach like HTCP, which repeatedly causes excess loss by using maximal acceleration when exceeding the capacity of the path)

(2) if a flow experiences high packet loss, slow down

So I would suggest something like:

"A congestion control algorithm should try to avoid causing excessively high rates of packet loss. To accomplish this, it should maintain its sending rate sufficiently near the recently measured capacity of the network path, and reduce its sending rate if experiencing high packet loss."

Reply:
I understand your point, but I am trying to find a way to say that without prescribing exactly the solution used in BBR. At the abstract level there are two variables, the sending rate and the size of bursts, and yes the proposed text misses the bursts. But many congestion control algorithms do not directly measure the "capacity of the network path" -- they will have to shrink the congestion window.

Reply:
I agree we don't want to prescribe exactly the solution used in BBR. But IMHO my proposed text does not do that.

Reno and CUBIC also use the approach: "maintain its sending rate sufficiently near the recently measured capacity of the network path". Though documents explaining the algorithms don't use this framework to explain the algorithms, they do explicitly track the recently measured capacity of the path (in volumetric terms, bytes or packets) using cwnd. And they maintain their sending rate sufficiently near the recently measured capacity by making changes in cwnd that growing cwnd no more than 2x per round trip.

If you still think that's too specific, how about:

"A congestion control algorithm should try to avoid causing excessively high rates of packet loss. To accomplish this, it should avoid excessive increases in sending rate, and reduce its sending rate if experiencing high packet loss."

Reply:
I like Neal's second proposal.  I might even delete the second sentence, to avoid presupposing anything about how a proposal meets the requirement. (I might add that "sending rate" can be used in a rate pacing context, or as a synonym for cwnd)

Reply:
Yes. Committing it now.